From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: tlauncher <team@totem.markets>
Date: Wed, 28 Oct 2026 05:25:46 +0530
Subject: [PATCH] tlauncher: install under /data/data/com.tux.tlauncher

nix-on-droid hard-codes the app's data dir in read-only options, so tlauncher
patches it instead of overriding it.
---
 modules/build/config.nix                  | 2 +-
 modules/user.nix                          | 2 +-
 pkgs/android-integration/termux-am.nix    | 2 +-
 pkgs/android-integration/termux-tools.nix | 8 ++++----
 4 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/modules/build/config.nix b/modules/build/config.nix
index ea971ee..95047d3 100644
--- a/modules/build/config.nix
+++ b/modules/build/config.nix
@@ -43,7 +43,7 @@ with lib;
 
   config = {
 
-    build.installationDir = "/data/data/com.termux.nix/files/usr";
+    build.installationDir = "/data/data/com.tux.tlauncher/files/usr";
 
   };
 
diff --git a/modules/user.nix b/modules/user.nix
index 6769e3d..0f02577 100644
--- a/modules/user.nix
+++ b/modules/user.nix
@@ -90,7 +90,7 @@ in
     };
 
     user = {
-      home = "/data/data/com.termux.nix/files/home";
+      home = "/data/data/com.tux.tlauncher/files/home";
     };
 
   };
diff --git a/pkgs/android-integration/termux-am.nix b/pkgs/android-integration/termux-am.nix
index 18c8a30..b21fb4b 100644
--- a/pkgs/android-integration/termux-am.nix
+++ b/pkgs/android-integration/termux-am.nix
@@ -3,7 +3,7 @@
 { stdenv, fetchFromGitHub, cmake }:
 
 let
-  appPath = "/data/data/com.termux.nix/files/apps/com.termux.nix";
+  appPath = "/data/data/com.tux.tlauncher/files/apps/com.tux.tlauncher";
   socketPath = "${appPath}/termux-am/am.sock";
 in
 stdenv.mkDerivation rec {
diff --git a/pkgs/android-integration/termux-tools.nix b/pkgs/android-integration/termux-tools.nix
index 7a4bf51..3976de0 100644
--- a/pkgs/android-integration/termux-tools.nix
+++ b/pkgs/android-integration/termux-tools.nix
@@ -25,8 +25,8 @@ stdenvNoCC.mkDerivation rec {
   patches = [ ./termux-tools.patch ];
   postPatch = ''
     substituteInPlace scripts/termux-setup-storage.in \
-      --replace @TERMUX_HOME@ /data/data/com.termux.nix/files/home/ \
-      --replace @TERMUX_APP_PACKAGE@ com.termux.nix
+      --replace @TERMUX_HOME@ /data/data/com.tux.tlauncher/files/home/ \
+      --replace @TERMUX_APP_PACKAGE@ com.tux.tlauncher
     substituteInPlace scripts/termux-open.in \
       --replace 'getopt ' '${getopt}/bin/getopt '
     substituteInPlace \
@@ -34,9 +34,9 @@ stdenvNoCC.mkDerivation rec {
       scripts/termux-wake-lock.in \
       scripts/termux-wake-unlock.in \
       --replace @TERMUX_APP_PACKAGE@.app com.termux.app \
-      --replace @TERMUX_APP_PACKAGE@ com.termux.nix
+      --replace @TERMUX_APP_PACKAGE@ com.tux.tlauncher
     substituteInPlace scripts/termux-reload-settings.in \
-      --replace @TERMUX_APP_PACKAGE@ com.termux.nix
+      --replace @TERMUX_APP_PACKAGE@ com.tux.tlauncher
     ${gnused}/bin/sed -i 's|^am |${termux-am}/bin/am |' scripts/*
 
     rm -r doc  # manpage is half misleading, pulling pandoc is not worth it
-- 
2.56.0

